### DISCUSSION TITLE
Support for brotlicffi in tests

### DISCUSSION THREAD

**Discussion by mgorny** [encode/httpx#2903] [Ideas]
While the package itself can work with either `brotli` or `brotlicffi`, `tests/test_decoders.py` explicitly requires `brotli`. We're currently working on having all packages support `brotlicffi` in Gentoo, since `brotli` doesn't work reliably on PyPy3. Could you please consider making the test accept `brotlicffi` as well?

I can make a patch if that helps.

**Comment by karpetrosyan** [comment]
Hi!

That makes perfect sense to me. If the implementation is `pypy`, we can just `import brotlicffi as brotli`.

> I can make a patch if that helps.

First, let's create an issue for this change.

**Reply by mgorny** [reply]
> That makes perfect sense to me. If the implementation is `pypy`, we can just `import brotlicffi as brotli`.

The `import`/`except` approach as used in `httpx/_compat.py` would be more convenient to us — in fact, I suppose ideally we could just `from httpx._compat import brotli`.

> First, let's create an issue for this change.

I'm sorry, are you suggesting that I create it or you do?

**Reply by karpetrosyan** [reply]
> I'm sorry, are you suggesting that I create it or you do?

I'll leave it to you :smiley:

**Reply by mgorny** [reply]
Done: https://github.com/encode/httpx/issues/2906
